controller-server/src/main/java/com/yahoo/vespa/hosted/controller/restapi/application/ApplicationApiHandler.java

```diff
@@ -347,9 +359,18 @@ else if (((Change.ApplicationChange)application.deploying().get()).revision().is
         for (URI rotation : rotations)
             globalRotationsArray.addString(rotation.toString());
 
-        // Deployments
+        // Deployments sorted according to deployment spec
```

Comment:
This is a lot of logic to add in ApplicationApiHandler.
I suggest moving it into the model - how about just let jobstatus() always return the jobs sorted?

Reply:
Agreed, will change.

Reply:
Right, I actually thought about doing that, but the problem is that you need access to deployment spec to sort and it's not available in `DeploymentJobs`.

Passing `DeploymentSpec` down to `DeploymentJobs` and storing a reference to it there just for the sake of sorting seems wrong. Suggestions?

Reply:
DeploymentJobs is owned by an Application, and cannot exist outside of an Application. You just need to pass the owning application to DeploymentJobs when it is created.
Then you can do owner.deploymentSpec() to access it.